Source/WebCore/rendering/RenderObject.cpp

```diff
@@ -110,18 +110,22 @@ RenderObject::SetLayoutNeededForbiddenScope::~SetLayoutNeededForbiddenScope()
 
 #endif
 
-struct SameSizeAsRenderObject : CanMakeWeakPtr<SameSizeAsRenderObject> {
+struct SameSizeAsRenderObject : WeakPtr<SameSizeAsRenderObject> {
```

Comment:
Why isn't this using `CanMakeCheckedPtr` ?

Reply:
I didn't mean to make this change, should remain `CanMakeWeakPtr< SameSizeAsRenderObject >`.

The issue we have the following data member below:
```
WeakHashSet<void*> cachedResourceClientAssociatedResources;
```

It comes from CachedImageClient, which is what RenderObject actually subclasses. But for some reason, we decided to have SameSizeAsRenderObject subclass `WeakPtr<SameSizeAsRenderObject>` instead of `CachedImageClient`. Instead we subclass `CanMakeWeakPtr` which `CachedImageClient` inherits and then include the CachedImageClient data member below. Not sure what's preferred. Let me know.

Reply:
So we have 2 possibilities, either:
1. We have `SameSizeAsRenderObject` subclass `CachedImageClient` and `CanMakeCheckedPtr`, like `RenderObject` does. And then remove the `cachedResourceClientAssociatedResources` member from `SameSizeAsRenderObject`.
2. Do what I did in this patch, keep subclassing `CanMakeWeakPtr` and add the data members inherited from `CanMakeCheckedPtr`, similarly to what was done with `cachedResourceClientAssociatedResources`.

Let me know which you prefer.